library/src/c_api/mod.rs

```diff
@@ -183,6 +194,23 @@ fn path_to_c_string(path: Option<PathBuf>) -> anyhow::Result<*mut c_char> {
     })
 }
 
+fn to_update_result(status: anyhow::Result<UpdateStatus>) -> UpdateResult {
+    let result = match status {
+        Ok(status) => {
+            let message = status.to_string();
+            return UpdateResult {
+                status: status as i32,
+                message: allocate_c_string(message.as_str()).unwrap(),
```

Comment:
`unwrap` will crash if it fails. We probably want `unwrap_or_else(|| std::null_ptr)` (not 100% sure on the syntax)